packages/mui-utils/src/useUndoAndRedo.ts

```diff
@@ -0,0 +1,55 @@
+import isEqual from 'lodash/isEqual';
+import { useState } from 'react';
+
+const useUndoAndRedo = <T>(initialValue?: T) => {
+  const [values, setValues] = useState<T[]>(initialValue !== undefined ? [initialValue] : []);
+  const [index, setIndex] = useState(initialValue !== undefined ? 0 : -1);
+
+  const currentValue = values[index];
+
+  const setValue = (value: T) => {
+    if (isEqual(currentValue, value)) {
```

Comment:
I think that we should compare by reference:

```suggestion
    if (currentValue === value) {
```